h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ParquetParser.java

```diff
@@ -79,6 +79,37 @@ public static ParseSetup guessSetup(ByteVec vec, byte[] bits) {
     return ppWriter.toParseSetup(metadataBytes);
   }
 
+  /**
+   * Overrides unsupported type conversions/mappings specified by the user.
+   * @param vec byte vec holding binary parquet data
+   * @param requestedTypes user-specified target types
+   * @return corrected types
+   */
+  public static byte[] correctTypeConversions(ByteVec vec, byte[] requestedTypes) {
+    byte[] metadataBytes = VecParquetReader.readFooterAsBytes(vec);
+    ParquetMetadata metadata = VecParquetReader.readFooter(metadataBytes, ParquetMetadataConverter.NO_FILTER);
+    byte[] roughTypes = roughGuessTypes(metadata.getFileMetaData().getSchema());
+    return correctTypeConversions(roughTypes, requestedTypes);
+  }
+
+  private static byte[] correctTypeConversions(byte[] roughTypes, byte[] requestedTypes) {
+    if (requestedTypes.length != roughTypes.length)
+      throw new IllegalArgumentException("Invalid column type specification: number of columns and number of types differ!");
+    byte[] resultTypes = new byte[requestedTypes.length];
+    for (int i = 0; i < requestedTypes.length; i++) {
+      if ((roughTypes[i] == Vec.T_NUM) || (roughTypes[i] == Vec.T_TIME)) {
+        // don't convert Parquet numeric/time type to non-numeric type in H2O
+        resultTypes[i] = roughTypes[i];
```

Comment:
I generate a warning instead which I think is a little bit more user friendly. I think requests of numeric->string conversions will be mostly accidental for data in binary formats.

It could be changed to an error too. This was motivated by a particular user who was struggling with our type specification ("all or nothing" can be hard if you have many columns).